spring-integration-core/src/main/java/org/springframework/integration/dsl/IntegrationComponentSpec.java

```diff
@@ -61,78 +68,83 @@ public final String getId() {
 
 	/**
 	 * @return the configured component.
+	 * @deprecated since 6.1 with no-op for end-user:
+	 * the {@link #getObject()} is called by the framework at the appropriate phase.
 	 */
+	@Deprecated(since = "6.1", forRemoval = true)
 	public T get() {
-		if (this.target == null) {
-			this.target = doGet();
-		}
-		return this.target;
+		return getObject();
 	}
 
 	@Override
 	public Class<?> getObjectType() {
-		return get().getClass();
+		return getObject().getClass();
 	}
 
+	/**
+	 * !!! This method must not be called from the target configuration !!!
+	 * @return the object backed by this factory bean.
+	 */
 	@Override
-	protected T createInstance() {
-		T instance = get();
-		if (instance instanceof InitializingBean) {
-			try {
-				((InitializingBean) instance).afterPropertiesSet();
-			}
-			catch (Exception e) {
-				throw new IllegalStateException("Cannot initialize bean: " + instance, e);
+	public T getObject() {
```

Comment:
Perhaps we should rename this to `internalGetObject()` with javadocs clearly specifying that it must only be called from framework components (or custom DSL specs for advanced users who might implement DSL extensions).

Then implement `getObject()` to throw an `UnsupportedOperationException`.

WDYT?

Reply:
Sorry, we cannot. The `IntegrationComponentSpec` can be declared as a bean, like any `FactoryBean`.
See our many examples in the tests like:
```
		@Bean
		public KafkaPublishSubscribeChannelSpec pubSub(KafkaTemplate<Integer, String> template,
				ConcurrentKafkaListenerContainerFactory<Integer, String> containerFactory) {

			return Kafka.publishSubscribeChannel(template, containerFactory, TEST_TOPIC7);
		}
```
In this case the target `IntegrationFlow` deals only with a result of this spec. Same way as it is done with regular `FactoryBean`.
That's why I pointed this in the doc.
At the same time the `IntegrationFlowBeanPostProcessor` is able to process a `ComponentsRegistration` from such a `IntegrationComponentSpec` in parallel.

Reply:
Oh; right; sorry.